sunpy/net/dataretriever/sources/gong.py

```diff
@@ -140,7 +149,7 @@ def download_from_nso(id, ObsID, time_range):
             elif (pattern_ == url_pattern_2):
                 urls = [url_pattern_2.format(ObsID=id[0].upper()) for id in instruments_to]
                 arr = [Scraper(pattern__).filelist(timerange) for pattern__ in urls]
-                result.extend([url for url in arr if len(url)>0])
+                [result.extend(url) for url in arr if len(url)>0]
```

Comment:
I think because your way would be incorrect. The ```extend``` method joins two arrays. With your way we are joining an array with an array of an array, the list comprehension gives an array which contains a single array inside, and we pass the final array (generated by the list comprehension). My way is generating all arrays within the list comprehension and passing them indiviually through the ```extend``` method. Also, when I made just your change, it was failing one of the tests - ```AttributeError: 'list' object has no attribute 'timeout'``` with this error message.